client/state/data-layer/wpcom/email-accounts/index.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * External Dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+import { convertToCamelCase } from 'state/data-layer/utils';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { EMAIL_ACCOUNTS_REQUEST } from 'state/action-types';
+import { errorNotice } from 'state/notices/actions';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import {
+	receiveGetEmailAccountsSuccess,
+	receiveGetEmailAccountsFailure,
+} from 'state/email-accounts/actions';
+import { registerHandlers } from 'state/data-layer/handler-registry';
+
+export const getEmailAccounts = ( action ) => {
+	if ( config.isEnabled( 'email-accounts/enabled' ) ) {
+		return receiveGetEmailAccountsSuccess( action.siteId, {
+			accounts: [
+				{
+					domain_name: 'domain.com',
+					product_name: 'Titan Basic',
+					product_slug: 'titan_basic',
+					product_type: 'titan',
```

Comment:
I though that way as well. Famously we have had products that have been preceded by their product type, but this is not always the case - I took a look at the products DB schema.

Since this is meant to be a generic schema for providers and one can't really tell how these things will be named in the future, I opted to be more specific.

Reply:
Oh I wasn't referring to the prefix - my bad - I should have been more specific. I was simply thinking that we could take advantage of functions that check if a product is from a certain type, e.g.:

```javascript
function getGSuiteAccounts( accounts ) {
  return accounts.filter( account => {
    return isGSuiteProductSlug( account.productSlug );
  } );
}
```